### PR TITLE
Profile pic

### PR DESCRIPTION
**Pull request by DiptoSarkar182**
User can upload profile picture. Had to move the json format to controller as I could not find way to include user profile_picture url in json response.

### CODE REVIEW

File: app/controllers/friend_lists_controller.rb [hunk 1]
```diff
@@ -6,6 +6,17 @@ def index
     @current_user_friend_lists = FriendList.current_user_friend_lists(current_user)
 
     if @current_user_friend_lists.any?
+      @current_user_friend_lists = @current_user_friend_lists.map do |friend|
+        profile_picture_url = friend.profile_picture.attached? ? url_for(friend.profile_picture) : nil
```
Comment (JakariaJishan) [inline app/controllers/friend_lists_controller.rb]: get profile_picture using .url method which is available for active storage.

**Review by JakariaJishan — CHANGES_REQUESTED** [review]
need changes

File: app/controllers/friend_lists_controller.rb (hunk 1 above)
Comment (DiptoSarkar182) [reply app/controllers/friend_lists_controller.rb]: added it.

File: app/controllers/users/registrations_controller.rb [hunk 2]
```diff
@@ -16,15 +16,28 @@ def destroy
...
-      }, status: :unprocessable_entity
+      if resource.update_with_password(resource_params)
+        render json: {
+          status: { code: 200, message: 'User information updated successfully.',
+                    data: UserSerializer.new(resource).serializable_hash[:data][:attributes] }
+        }, status: :ok
+      else
+        render json: {
+          status: { message: "User information couldn't be updated. #{resource.errors.full_messages.to_sentence}" }
+        }, status: :unprocessable_entity
```
Comment (JakariaJishan) [inline app/controllers/users/registrations_controller.rb]: try to reduce nesting if else conditions.

File: app/models/challenge_friend.rb [hunk 3]
```diff
@@ -33,8 +34,34 @@ def self.send_challenge(user, params)
...
+          friend_id: friend.id
+        }
+      ).where(
+        challenge_type: 'global',
+        status: 'pending'
+      ).exists?
+
+      return { message: 'A pending global challenge already exists between you and the user', status: :unprocessable_entity } if existing_pending_global_challenge
+    else
+      return { message: 'Invalid challenge type', status: :unprocessable_entity }
```
Comment (JakariaJishan) [inline app/models/challenge_friend.rb]: try to reduce nesting if else conditions. Also change the queries.

File: app/models/challenge_friend.rb [hunk 4]
```diff
@@ -76,5 +103,46 @@ def self.cancel_challenge(user, challenge_id)
...
+
+    return { message: 'Challenge not found or not eligible for acceptance', status: :not_found } if challenge.nil?
+
+    if user.coins < challenge.amount_of_betting_coin
+      return { message: 'Insufficient coins to accept this challenge', status: :unprocessable_entity }
+    end
+
+    ActiveRecord::Base.transaction do
+      user.update!(coins: user.coins - challenge.amount_of_betting_coin)
+      challenge.update!(status: 'accepted')
```
Comment (JakariaJishan) [inline app/models/challenge_friend.rb]: try to use .increament! and .decrement method

**Review by JakariaJishan — CHANGES_REQUESTED** [review]
please address all comments

File: app/controllers/users/registrations_controller.rb (hunk 2 above)
Comment (DiptoSarkar182) [reply app/controllers/users/registrations_controller.rb]: reduced the nesting if-else.

File: app/models/challenge_friend.rb (hunk 3 above)
Comment (DiptoSarkar182) [reply app/models/challenge_friend.rb]: removed if-else and changed queries.

File: app/models/challenge_friend.rb (hunk 4 above)
Comment (DiptoSarkar182) [reply app/models/challenge_friend.rb]: increment_coins and decrement_coins added in user model.

**Review by JakariaJishan — APPROVED** [review]
Congrats....Approved..

**Pull request merged**